### PR TITLE
wait for prometheus-pushgateway to be available

### PR DESCRIPTION
**Pull request by coryschwartz**
Fixes https://github.com/ipfs/testground/issues/685

Fixes https://github.com/ipfs/testground/issues/598

Fixes https://github.com/ipfs/testground/issues/637

### CODE REVIEW

File: sdk/runtime/runner.go [hunk 1]
```diff
@@ -20,11 +21,33 @@ func Invoke(tc func(*RunEnv) error) {
...
 	go func() {
 		pushInterval := 5 * time.Second
+
+		// Wait until the pushgateway is ready.
+		runenv.RecordMessage("Waiting for pushgateway to become accessible.")
+		var resbuf []byte
+		for {
+			select {
+			case <-time.After(pushInterval):
+				resp, err := http.Get("http://prometheus-pushgateway:9091/-/ready")
```
Comment (coryschwartz) [inline sdk/runtime/runner.go]: I *definitely* want to get rid of the hard-coded urls everywhere. Should turn this into an env variable, like the REDIS_HOST, I think.

Comment (raulk) [reply sdk/runtime/runner.go]: Yes, go for it.

* `cluster:k8s` and `local:docker` runners would inject `prometheus-pushgateway:9091`
* `local:exec` should inject `localhost:9091`.

Comment (raulk) [reply sdk/runtime/runner.go]: Oh, nice. I hadn't realised that pushgateway had a `ready` endpoint.

Comment (raulk) [reply sdk/runtime/runner.go]: How does this interact with the defer below, which actually forces a final push? We should probably capture the error there and make it less scary, with a nicer message like: `metrics push failed during instance shutdown; prometheus metrics from this instance may not be available; err: %s`

Comment (coryschwartz) [reply sdk/runtime/runner.go]: it does print the final message " error during end metric push...", right now. But of course that could be changed.

Comment (coryschwartz) [reply sdk/runtime/runner.go]: I created https://github.com/ipfs/testground/issues/711 to change this into a environment-delivered parameter.

**Pull request merged**